### ISSUE TITLE
USB test no longer works on Imix

### ISSUE THREAD

**Issue by bradjc** [tock/tock#1891]
I tried running `libtock-c/examples/tests/usb` on Imix, and get the following panic:

```
Initialization complete. Entering main loop


Kernel panic at chips/sam4l/src/usbc/mod.rs:1443:
	"Bad endpoint buffer size"
	Kernel version release-1.5-300-g926a0b221

---| No debug queue found. You can set it with the DebugQueue component.

---| Fault Status |---
No faults detected.

---| App Status |---
App: usb   -   [Running]
 Events Queued: 0   Syscall Count: 6   Dropped Callback Count: 0
 Restart Count: 0
 Last Syscall: Some(COMMAND { driver_number: 131077, subdriver_number: 1, arg0: 0, arg1: 0 })

 ╔═══════════╤══════════════════════════════════════════╗
 ║  Address  │ Region Name    Used | Allocated (bytes)  ║
 ╚0x20008000═╪══════════════════════════════════════════╝
             │ ▼ Grant        1080 |   1080
  0x20007BC8 ┼───────────────────────────────────────────
             │ Unused
  0x20006A5C ┼───────────────────────────────────────────
             │ ▲ Heap            0 |   4460               S
  0x20006A5C ┼─────────────────────────────────────────── R
             │ Data            604 |    604               A
  0x20006800 ┼─────────────────────────────────────────── M
             │ ▼ Stack          64 |   2048
  0x200067C0 ┼───────────────────────────────────────────
             │ Unused
  0x20006000 ┴───────────────────────────────────────────
             .....
  0x00042000 ┬─────────────────────────────────────────── F
             │ App Flash      8152                        L
  0x00040028 ┼─────────────────────────────────────────── A
             │ Protected        40                        S
  0x00040000 ┴─────────────────────────────────────────── H

  R0 : 0x00020005    R6 : 0x00041824
  R1 : 0x00000001    R7 : 0x20006000
  R2 : 0x00000000    R8 : 0x00000000
  R3 : 0x00000000    R10: 0x00000000
  R4 : 0x200067E0    R11: 0x00000000
  R5 : 0x20006800    R12: 0xACE5978D
  R9 : 0x20006800 (Static Base Register)
  SP : 0x200067C0 (Process Stack Pointer)
  LR : 0x00040235
  PC : 0x000401CE
 YPC : 0x000401CE

 APSR: N 0 Z 0 C 1 V 0 Q 0
       GE 0 0 0 0
 EPSR: ICI.IT 0x00
       ThumbBit true

 Cortex-M MPU
  Region 0: [0x20006000:0x20008000], length: 8192 bytes; ReadWrite (0x3)
    Sub-region 0: [0x20006000:0x20006400], Enabled
    Sub-region 1: [0x20006400:0x20006800], Enabled
    Sub-region 2: [0x20006800:0x20006C00], Enabled
    Sub-region 3: [0x20006C00:0x20007000], Disabled
    Sub-region 4: [0x20007000:0x20007400], Disabled
    Sub-region 5: [0x20007400:0x20007800], Disabled
    Sub-region 6: [0x20007800:0x20007C00], Disabled
    Sub-region 7: [0x20007C00:0x20008000], Disabled
  Region 1: [0x00040000:0x00042000], length: 8192 bytes; ReadOnly (0x6)
    Sub-region 0: [0x00040000:0x00040400], Enabled
    Sub-region 1: [0x00040400:0x00040800], Enabled
    Sub-region 2: [0x00040800:0x00040C00], Enabled
    Sub-region 3: [0x00040C00:0x00041000], Enabled
    Sub-region 4: [0x00041000:0x00041400], Enabled
    Sub-region 5: [0x00041400:0x00041800], Enabled
    Sub-region 6: [0x00041800:0x00041C00], Enabled
    Sub-region 7: [0x00041C00:0x00042000], Enabled
  Region 2: Unused
  Region 3: Unused
  Region 4: Unused
  Region 5: Unused
  Region 6: Unused
  Region 7: Unused

To debug, run `make debug RAM_START=0x20006000 FLASH_INIT=0x40051`
in the app's folder and open the .lst file.
```

That tracks back to this check failing:

https://github.com/tock/tock/blob/0054c97dcfdd1e0b4b768dabbf586008561bc550/chips/sam4l/src/usbc/mod.rs#L1441-L1447

I changed `ctrl_buffer` to a `Buffer8`:

https://github.com/tock/tock/blob/0054c97dcfdd1e0b4b768dabbf586008561bc550/capsules/src/usb/usbc_client_ctrl.rs#L29-L37

and it no longer panics:

```
Initialization complete. Entering main loop
USB test: Enabled and attached
Bus reset
Bus reset
Bus reset
Bus reset
Bus reset
Bus reset
Bus reset
Bus reset
Bus reset
Bus reset
Bus reset
Bus reset
Bus reset
Bus reset
Bus reset
Bus reset
Bus reset
Bus reset
```

but the tests in tools/usb don't work, for example:

```
/tools/usb/control_test:
 $ cargo run
    Finished dev [unoptimized + debuginfo] target(s) in 0.02s
     Running `target/debug/control-test`
Searching for device ...
   20:06 Vendor:0403 Product:6015
   26:03 Vendor:05ac Product:8510
   29:04 Vendor:0a5c Product:4500
   29:03 Vendor:05ac Product:0262
   29:02 Vendor:0424 Product:2512
   29:01 Vendor:8087 Product:0024
   26:01 Vendor:8087 Product:0024
thread 'main' panicked at 'Matching device not found', src/main.rs:57:18
note: run with `RUST_BACKTRACE=1` environment variable to display a backtrace
```

**Comment by bradjc** [comment]
For my own reference, 9e6b2f622da27a3f8324e228761e426ca1e1bb28 appears to be a working commit.

**Comment by bradjc** [comment]
Fixes in PRs.

**Comment by bradjc** [comment]
Fixed with #1902.
